File: src/main/java/dev/retrotv/utils/SecureRandomUtil.java
package dev.retrotv.utils;

import java.security.SecureRandom;

public class SecureRandomUtil {

    private SecureRandomUtil() {
        throw new IllegalStateException("유틸리티 클래스 입니다.");
    }

    public static byte[] generate(int len) {
        SecureRandom sr = new SecureRandom();
        byte[] randomData = new byte[len];
        sr.nextBytes(randomData);

        return randomData;
    }
}

File: src/main/kotlin/dev/retrotv/utils/SecureRandomUtil.kt
package dev.retrotv.utils

import java.security.SecureRandom

class SecureRandomUtil private constructor() {
    init {
        throw IllegalStateException("유틸리티 클래스 입니다.")
    }

    companion object {
        fun generate(len: Int): ByteArray {
            val sr = SecureRandom()
            val randomData = ByteArray(len)
            sr.nextBytes(randomData)
            return randomData
        }
    }
}
